### PR TITLE
[nodes] Add Edge data type

### PR DESCRIPTION
**Pull request by Kyle0654**
Adds an `Edge` data type, replacing the current tuple used for edges.

### CODE REVIEW

**Review by blessedcoolant — APPROVED** [review]

**Comment by blessedcoolant** [comment]
Code reads fine. Seems pretty straightforward. I'll let @psychedelicious have a look at it too before I merge it.

**Review by psychedelicious — APPROVED** [review]
Tested on CLI and UI. Also confirm this fixes the API client/types generation.

**Pull request merged**
